src/Redux/todo-reducer.ts

```diff
@@ -5,68 +5,72 @@ export type TodoActionsType = InferActionTypes<typeof todoActions>
 export type TodoListItemType = {
     id: number,
     task: string,
-    progress: boolean
+    progress: boolean,
 }
 
 export type TodoInitialStateType = {
-    todoList: TodoListItemType[]
-    progressAll: boolean 
-    changedFilter: string
+    todoList: TodoListItemType[],
+    progressAll: boolean,
+    changedFilter: string,
+    namesFilters: string[],
 }
 
 const initialState: TodoInitialStateType = {
     todoList: [],
     progressAll: false,
     changedFilter: 'all',
+    namesFilters: ['all', 'active', 'completed'],
+}
+
+const getIndexById = (id: number, array: TodoListItemType[]) => {
+    const index = array.findIndex(element => {
+        return element.id === id
+    })
+    return index
+}
+enum ActionTypes {
+    ADD_TASK = 'ADD-TASK',
+    REMOVE_TASK = 'REMOVE-TASK',
+    TOGGLE_PROGRESS = 'TOGGLE-PROGRESS',
+    CHANGE_TASK = 'CHANGE-TASK',
+    TOGGLE_PROGRESS_ALL = 'TOGGLE-PROGRESS-ALL',
+    REMOVE_TASKS_ALL_COMPLITED = 'REMOVE-TASKS-ALL-COMPLITED',
+    CHANGE_FILTER = 'CHANGE-FILTER',
 }
 
 export const todoReducer = (state = initialState, action: TodoActionsType ): TodoInitialStateType => {
     switch (action.type) {
-        case 'ADD-TASK':
+        case ActionTypes.ADD_TASK:
             return {...state, todoList: [action.payload, ...state.todoList] }
-        case 'REMOVE-TASK':
+        case ActionTypes.REMOVE_TASK:
             return {...state, todoList: state.todoList.filter(task => {
                 return task.id !== action.payload
             })}
-        case 'TOGGLE-PROGRESS':
-            return {...state, todoList: state.todoList.map(task => {
-                if(task.id === action.payload) {
-                    task.progress = !task.progress
-                }
-                return task
-            })}
-        case 'CHANGE-TASK': 
+        case ActionTypes.TOGGLE_PROGRESS:
+            let index = getIndexById(action.payload, state.todoList)
+            let currentProgress = state.todoList[index].progress
+            state.todoList[index].progress = !currentProgress
+            return {...state, todoList: [...state.todoList ]}
```

Comment:
Here:
1. Don't use let when you can use const
2. You need to create a new array of items, not mutate directly (state.todoList[index]...)
3. findIndex can return -1, you don't have a check for this case